src/game/common/modules/behaviormodule.h

```diff
@@ -116,18 +116,17 @@ class BehaviorModule : public ObjectModule, public BehaviorModuleInterface
     IMPLEMENT_ABSTRACT_POOL(BehaviorModule)
 
 protected:
+    BehaviorModule(Thing *thing, const ModuleData *module_data);
```

Comment:
ctors were public

Reply:
I think it is better to keep these protected unless explicitly required. These modules don't look like they want to be created and destroyed just like that, but are created through the `Friend_New_Module_Instance` functions. See `SwayClientUpdate::Friend_New_Module_Instance` for example.